### PR TITLE
[chip-tool][ICD] Store ICDClientInfo when running icd registration independently

### PR DESCRIPTION
**Pull request by yunhanw-google**

### CODE REVIEW

File: examples/chip-tool/commands/clusters/ModelCommand.cpp [hunk 1]
```diff
@@ -85,6 +85,22 @@ void ModelCommand::ClearICDEntry(const chip::ScopedNodeId & nodeId)
...
+    {
+        err = CHIPCommand::sICDClientStorage.StoreEntry(clientInfo);
+    }
+
+    if (err != CHIP_NO_ERROR)
+    {
+        CHIPCommand::sICDClientStorage.RemoveKey(clientInfo);
+        ChipLogError(chipTool, "Failed to persist symmetric key with error: %" CHIP_ERROR_FORMAT, err.Format());
+        return;
+    }
```
Comment (mkardous-silabs) [inline examples/chip-tool/commands/clusters/ModelCommand.cpp:101]: this logic can be simplified to only have 1 conditionnal block. either with an if else or with if(...return)...

File: examples/chip-tool/commands/clusters/ClusterCommand.h [hunk 2]
```diff
@@ -117,7 +127,28 @@ class ClusterCommand : public InteractionModelCommands, public ModelCommand, pub
...
+                    ChipLogError(chipTool, "Failed to decode ICD counter: %" CHIP_ERROR_FORMAT, err.Format());
+                    return;
+                }
+
+                chip::app::ICDClientInfo clientInfo;
+                clientInfo.peer_node         = mScopedNodeId;
+                clientInfo.monitored_subject = mMonitoredSubject;
+                clientInfo.start_icd_counter = value.ICDCounter;
+
+                ModelCommand::AddICDEntry(clientInfo, chip::ByteSpan(mICDSymmetricKey));
```
Comment (bzbarsky-apple) [inline examples/chip-tool/commands/clusters/ClusterCommand.h:148]: Why does this need the `ModelCommand` prefix?

File: examples/chip-tool/commands/clusters/ModelCommand.cpp [hunk 3]
```diff
@@ -85,6 +85,22 @@ void ModelCommand::ClearICDEntry(const chip::ScopedNodeId & nodeId)
     }
 }
 
+void ModelCommand::AddICDEntry(chip::app::ICDClientInfo & clientInfo, chip::ByteSpan key)
```
Comment (bzbarsky-apple) [inline examples/chip-tool/commands/clusters/ModelCommand.cpp:88]: Why is that first arg non-const?

File: examples/chip-tool/commands/clusters/ModelCommand.h [hunk 4]
```diff
@@ -69,6 +69,7 @@ class ModelCommand : public CHIPCommand
 
     virtual void ClearICDEntry(const chip::ScopedNodeId & nodeId);
 
+    virtual void AddICDEntry(chip::app::ICDClientInfo & clientinfo, chip::ByteSpan key);
```
Comment (bzbarsky-apple) [inline examples/chip-tool/commands/clusters/ModelCommand.h:72]: Why is this virtual?  It doesn't need to be virtual.

It looks like ClearICDEntry is also incorrectly virtual....

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: examples/chip-tool/commands/clusters/ModelCommand.cpp (hunk 3 above)
Comment (yunhanw-google) [reply examples/chip-tool/commands/clusters/ModelCommand.cpp:88]: sICDClientStorage.SetKey is happening inside this function,
just rename this function to StoreICDEntryWithKey to reflect this. thanks

File: examples/chip-tool/commands/clusters/ClusterCommand.h (hunk 2 above)
Comment (yunhanw-google) [reply examples/chip-tool/commands/clusters/ClusterCommand.h:148]: removed, thanks

File: examples/chip-tool/commands/clusters/ModelCommand.h (hunk 4 above)
Comment (yunhanw-google) [reply examples/chip-tool/commands/clusters/ModelCommand.h:72]: removed, thanks

File: examples/chip-tool/commands/clusters/ModelCommand.cpp (hunk 1 above)
Comment (yunhanw-google) [reply examples/chip-tool/commands/clusters/ModelCommand.cpp:101]: we need check the error from CHIPCommand::sICDClientStorage.StoreEntry, and we still need two blocks?

**Review by bzbarsky-apple — APPROVED** [review]

**Review by joonhaengHeo — APPROVED** [review]

**Pull request merged**